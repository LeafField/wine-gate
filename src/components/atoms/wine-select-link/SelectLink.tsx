import React, { FC } from "react";
import Link from "next/link";
import Image from "next/image";
import WineImage from "../../../images/wine_select_image.jpg";

const SelectLink: FC = () => {
  return (
    <Link
      className="group relative col-span-2 block h-[12rem] w-full overflow-hidden tablet:h-[24.0625rem]"
      href={"/select"}
    >
      <figure className="absolute h-full w-full transition-transform duration-300 group-hover:scale-110">
        <Image
          src={WineImage}
          alt="棚に並ぶワインボトル"
          fill
          style={{ objectFit: "cover" }}
        />
      </figure>
      {/* <div className="z-front relative flex h-full w-full bg-gradient-to-t from-black/80 to-transparent align-bottom"> */}
      <div className="z-front bg-image-cover relative flex h-full w-full align-bottom">
        <h2 className="mt-auto w-full px-6 py-4 text-end text-heading3 text-white">
          簡単なワインの選び方
        </h2>
      </div>
    </Link>
  );
};

export default SelectLink;
