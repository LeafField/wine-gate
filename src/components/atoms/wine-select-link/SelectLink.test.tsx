import { screen, render } from "@testing-library/react";
import SelectLink from "./SelectLink";

describe("SelectLinkの単体テスト", () => {
  test("SelectLinkが存在することを確認する", () => {
    render(<SelectLink />);
    expect(screen.getByText("簡単なワインの選び方")).toBeInTheDocument();
  });
  test("スナップショットテスト", () => {
    const { container } = render(<SelectLink />);
    expect(container).toMatchSnapshot();
  });
});
