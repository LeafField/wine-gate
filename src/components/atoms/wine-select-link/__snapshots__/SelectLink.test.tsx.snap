// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectLinkの単体テスト スナップショットテスト 1`] = `
<div>
  <a
    class="group relative col-span-2 block h-[12rem] w-full overflow-hidden tablet:h-[24.0625rem]"
    href="/select"
  >
    <figure
      class="absolute h-full w-full transition-transform duration-300 group-hover:scale-110"
    >
      <img
        alt="棚に並ぶワインボトル"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
      />
    </figure>
    <div
      class="z-front bg-image-cover relative flex h-full w-full align-bottom"
    >
      <h2
        class="mt-auto w-full px-6 py-4 text-end text-heading3 text-white"
      >
        簡単なワインの選び方
      </h2>
    </div>
  </a>
</div>
`;
